homeassistant/components/fjaraskupan/__init__.py

```diff
@@ -0,0 +1,124 @@
+"""The Fjäråskupan integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from bleak import BleakScanner
+from bleak.backends.device import BLEDevice
+from bleak.backends.scanner import AdvertisementData
+from fjaraskupan import DEVICE_NAME, Device, State
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DISPATCH_DETECTION, DOMAIN
+
+PLATFORMS = ["fan"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class EntryState:
+    """Store state of config entry."""
+
+    scanner: BleakScanner
+    device: Device
+    coordinator: DataUpdateCoordinator[State]
+    device_info: DeviceInfo
+
+
+async def async_startup_scanner(hass):
+    """Startup the background event scanner."""
+    scanner = BleakScanner()
+    await scanner.start()
```

Comment:
I agree that we should consider making this an integration. Prefer if we can set it up in a way to allow different implementations in the future because I want to start using ESPHome nodes as Bluetooth remote clients eventually.

Reply:
Bleak works on osx, Linux and windows. So it's a rather good base. We don't really need any other implementation i think.

Reply:
I agree that we don't need an alternative local implementation. But the problem is that a) houses can be big and b) Home Assistant might live in the basement. In those cases having ESP32 devices scattered through the house to connect to things via Bluetooth would be great.

Reply:
Right indeed. Struggle with that myself. We could maybe push a network backend to upstream bleak.

Reply:
That would be even better